### PR TITLE
[v7r0] Modify codes to support Python 3

### PR DESCRIPTION
**Pull request by radonys**
BEGINRELEASENOTES

*WMS
CHANGE: Modify codes using `python-modernize` to support Python3 along with Python2.

*Core
CHANGE: Modify codes using `python-modernize` to support Python3 along with Python2.

ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
This is of course good, but we need to discuss corner cases and how to proceed in the next BiLD meeting. For example, there are also several calls to `dict.items()` that should be treated.

File: WorkloadManagementSystem/DB/JobDB.py [hunk 1]
```diff
@@ -796,135 +799,135 @@ def setJobParameters( self, jobID, parameters ):
...
     """
 
     # Extract initital job parameters
     parameters = {}
-    if classadJob.lookupAttribute( "Parameters" ):
-      parameters = classadJob.getDictionaryFromSubJDL( "Parameters" )
-    res = self.setJobParameters( jobID, parameters.items() )
+    if classadJob.lookupAttribute("Parameters"):
+      parameters = classadJob.getDictionaryFromSubJDL("Parameters")
+    res = self.setJobParameters(jobID, list(parameters.items()))
```
Comment (andresailer) [inline WorkloadManagementSystem/DB/JobDB.py]: Was this change done automatically? Cf. https://github.com/DIRACGrid/DIRAC/pull/3763#discussion_r205439203

Comment (radonys) [reply WorkloadManagementSystem/DB/JobDB.py]: Yes, enclosing `parameters.items()` in a `list` was automatically done with the `python-modernize` tool.

Comment (fstagni) [reply WorkloadManagementSystem/DB/JobDB.py]: Then I don't fully get it: at line 1240 of this same module there is `for param, val in voPolicy.items():` and that line was not touched. It seems that when there's a loop on `.items()` results, python-modernize ignores it. This is confusing and IMHO dangerous.

Comment (radonys) [reply WorkloadManagementSystem/DB/JobDB.py]: Yes, that seems weird (regarding changes). Let me explore a bit on this and can be discussed later.

File: WorkloadManagementSystem/Service/JobStateUpdateHandler.py [hunk 2]
```diff
@@ -57,7 +60,7 @@ def initialize(self):
     return S_OK()
 
   ###########################################################################
-  types_updateJobFromStager = [[basestring, int, long], basestring]
+  types_updateJobFromStager = [[six.string_types, int, int], six.string_types]
```
Comment (atsareg) [inline WorkloadManagementSystem/Service/JobStateUpdateHandler.py]: [six.string_types, int, int] <- why int, int ? typo ? should be kept int, long, I guess

Comment (fstagni) [reply WorkloadManagementSystem/Service/JobStateUpdateHandler.py]: There is no `long` type in python 3

Comment (atsareg) [reply WorkloadManagementSystem/Service/JobStateUpdateHandler.py]: Then it should be just [six.string_types, int]

Comment (radonys) [reply WorkloadManagementSystem/Service/JobStateUpdateHandler.py]: @atsareg Actually these changes are automated by the `python-modernize` tool. Some of these changes need to be discussed due to such issues.

**Comment by fstagni** [comment]
Hi @radonys the branch has conflicts, could you please solve them?

**Comment by radonys** [comment]
@fstagni All conflicts resolved now.

File: WorkloadManagementSystem/Service/JobStateUpdateHandler.py (hunk 2 above)
Comment (fstagni) [reply WorkloadManagementSystem/Service/JobStateUpdateHandler.py]: No need to repeat `int, int`: just once is enough. This is valid here and in the following lines.

**Comment by fstagni** [comment]
I will merge this PR as soon as the check from Travis is OK (just started now)

**Pull request merged**